### PR TITLE
Update: Replace route API and remove OAuth from Navigate in AR

### PR DESCRIPTION
**Pull request by yo1995**
This PR resolves the issue in

- common-samples/issues/2678
- common-samples/issues/2677

that uses API key instead of OAuth to authenticate the user for the navigation task.

### CODE REVIEW

**Comment by yo1995** [comment]
Feedback:
- We should also describe why and how to use API key, and add a link to the https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/#available-services section

**Review by vquach2404 — DISMISSED** [review]

**Comment by yo1995** [comment]
@vquach2404 Do you have any suggestion on the README?

Sarat mentioned that users who are not familiar with the API keys might be confused if they copy and paste this sample to set up their own project. Thus I made changes to the README to explain, but don't know if the new instructions convey.

File: arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md [hunk 1]
```diff
@@ -52,13 +52,17 @@ When you start, route instructions will be displayed and spoken. As you proceed
 
 This sample uses Esri's [World Elevation](https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer) service to ensure that route lines are placed appropriately in 3D space. It uses Esri's [World Route](https://www.arcgis.com/home/item.html?id=1feb41652c5c4bd2ba5c60df2b4ea2c4) service to calculate routes. The world routing service requires authentication and does consume ArcGIS Online credits.
 
-**Note:** This item requires an ArcGIS Online organizational subscription or an ArcGIS Developer account and consumes credits.
+**Note:** This routing service requires one of the following authentication methods
 
-To access this sample, you'll need to do one of the following:
+* An ArcGIS Online organizational subscription and consumes credits
```
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md:57]:
```suggestion
* An ArcGIS Online organizational subscription and credits
```
Hmm, seems like "consumes credits" is being used as a noun here but to my knowledge, there isn't a proper term for Esri credits. I'd suggest just leaving it like this.

File: arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md [hunk 2]
```diff
@@ -52,13 +52,17 @@ When you start, route instructions will be displayed and spoken. As you proceed
...
+* An ArcGIS Developer account and consumes credits
```
Comment (vquach2404) [inline arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md:58]:
```suggestion
* An ArcGIS Developer account and credits
```

File: arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md [hunk 3]
```diff
@@ -52,13 +52,17 @@ When you start, route instructions will be displayed and spoken. As you proceed
...
+* An API key
 
-* Sign in with an account that is a member of an organizational subscription.
-* Sign in with a developer account.
-* Register an application and use your application's credentials.
+To access this sample with ArcGIS identity/named user login using OAuth workflow, please read more at [ArcGIS Identity](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/arcgis-identity/).
+
+The routing service can also be accessed with an API key, either specified in AppDelegate or in the sample. Please read more at [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/).
```
Comment (philium) [inline arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/README.md:63]: This reference two things that users may not have: 1) `AppDelegate` (their app delegate class may not be named that or they may not have an app delegate if using SwiftUI), and 2) the sample (their app is not going to be a sample viewer). Could that part just be taken out?

```suggestion
The routing service can also be accessed with an API key. Please read more at [API keys](https://developers.arcgis.com/documentation/mapping-apis-and-services/security/api-keys/).
```

**Review by vquach2404 — COMMENTED** [review]
I think the changes to the README explain the situation well. I made a few suggestions to make it slightly more clear.

**Comment by yo1995** [comment]
Thanks for the suggestions!

**Review by vquach2404 — APPROVED** [review]

**Review by philium — APPROVED** [review]

**Review by saratchandrakarumuri — APPROVED** [review]

**Pull request merged**